plugins/generator-1.17.1/forge-1.17.1/templates/procedure.java.ftl

```diff
@@ -33,29 +33,49 @@ package ${package}.procedures;
 
 import net.minecraftforge.eventbus.api.Event;
 
+<#assign nullableDependencies = []/>
+<#list dependencies as dependency>
+	<#if dependency.getType(generator.getWorkspace()) != "double"
+		&& dependency.getType(generator.getWorkspace()) != "LevelAccessor"
+		&& dependency.getType(generator.getWorkspace()) != "ItemStack"
+		&& dependency.getType(generator.getWorkspace()) != "BlockState"
+		&& dependency.getType(generator.getWorkspace()) != "InteractionResult"
+		&& dependency.getType(generator.getWorkspace()) != "boolean">
+		<#assign nullableDependencies += [dependency.getName()]/>
+	</#if>
+</#list>
+
+<#compress>
+
 <#if trigger_code?has_content>
 ${trigger_code}
 <#else>
 public class ${name}Procedure {
 </#if>
 
-	public static <#if return_type??>${return_type.getJavaType(generator.getWorkspace())}<#else>void</#if> execute(Map<String, Object> dependencies) {
+	<#if trigger_code?has_content>
+	public static <#if return_type??>${return_type.getJavaType(generator.getWorkspace())}<#else>void</#if> execute(
 		<#list dependencies as dependency>
-			if(dependencies.get("${dependency.getName()}") == null) {
-				if(!dependencies.containsKey("${dependency.getName()}"))
-					${JavaModName}.LOGGER.warn("Failed to load dependency ${dependency.getName()} for procedure ${name}!");
-				<#if return_type??>return ${return_type.getDefaultValue(generator.getWorkspace())}<#else>return</#if>;
-			}
-        </#list>
+			${dependency.getType(generator.getWorkspace())} ${dependency.getName()}<#if dependency?has_next>,</#if>

```

Comment:
Minor thing, but you can use `<#sep>,` instead of `<#if dependency?has_next>,</#if>`

Reply:
We don't use this in the other templates, so maybe we should keep the same thing everywhere.

Reply:
We do, in the 1.17 bounding boxes file

Reply:
Didn't know about it, neat, will use that